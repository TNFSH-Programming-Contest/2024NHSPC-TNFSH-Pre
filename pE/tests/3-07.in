2562
984922 902996 955855 961232 933975 955358 990016 913283 984882 958177 912150 945938 989034 978017 923905 919975 990297 959189 918162 959840 926081 905241 953818 957923 971292 905863 982965 927528 963750 991263 993479 919476 915209 958967 955132 929816 970324 916582 997332 953589 978236 988607 981322 921599 964194 976689 988946 956670 998092 937450 916809 922230 950412 937134 914465 915471 919056 953960 954356 906498 925200 961807 968324 905268 988792 967238 950308 961541 930390 947900 924659 942102 932021 969034 960016 928733 935505 987531 920182 961181 940611 982331 954240 969139 960264 950224 979637 902036 953123 948643 996952 978061 951372 964857 959143 952328 964218 997263 913212 991192 981840 953786 983132 906733 969358 954608 998148 978088 945706 900176 967288 965010 938350 925108 931172 946011 986546 926482 996742 968489 951363 937851 999097 971399 971640 964682 929989 959933 995842 994169 950048 994746 901511 914708 904225 997982 929764 961524 915847 990092 969225 923203 902984 929303 947817 943447 928943 922916 920808 906790 959916 906602 909447 917183 920850 908472 921939 938193 963544 988993 912231 984674 987777 970073 930815 991210 981632 961244 998981 968095 962999 988346 935015 969357 930826 941899 913702 973651 989850 935052 996257 933593 927649 995233 952164 902181 983769 909625 992603 966620 988250 954646 982020 990661 972240 996784 943471 953711 919227 918353 918395 992935 905121 933404 967379 934395 941572 991413 986703 956595 993289 932376 904355 926960 986047 925493 941012 985987 976135 935324 951348 917132 929680 997460 905484 989787 911898 994378 951877 945349 941641 945334 963845 991132 998700 983862 998790 935961 923858 990469 961463 939967 911979 957701 912813 926908 996307 903631 961451 927414 975136 983699 923525 955979 968016 935836 958942 911673 910925 904101 988079 979386 928450 921529 962547 964835 976789 921575 929067 986981 952872 933940 992615 948219 998225 917777 915143 953228 966482 900833 900693 995381 938319 935172 947456 949060 946669 999083 913511 991004 969093 914743 902687 948447 994445 944938 934874 940481 929790 988682 957099 994396 973841 982046 983752 995424 989648 943038 918855 905003 955223 960989 975596 986274 996528 933113 979304 913141 974566 930901 901017 972269 945255 930706 907661 921160 913762 970727 913570 957365 961761 999590 975532 953522 904486 969126 926988 972170 948993 925785 992127 952298 938080 948215 901946 960868 958580 977982 953362 927050 919483 901993 956829 956370 908775 914145 932033 909145 918501 985872 993085 948253 944009 955986 988758 961068 904734 936199 920213 981890 974871 944906 965784 960347 933758 935674 990493 926181 971808 972518 946735 928741 953080 941751 986033 919645 915630 942479 940071 909008 955760 983193 963568 903334 982076 953380 991278 904037 952485 969040 998149 954936 958887 996117 930939 984022 958260 993278 975500 956106 975147 991214 923282 964574 909260 962576 982203 952414 914578 928407 952536 974233 928997 900477 937588 966624 966385 929013 939820 990552 926552 987435 943554 980118 909068 940688 963042 954004 960240 982453 977922 906107 933049 980195 976346 950861 983358 989177 911927 926470 926105 912989 921042 936708 966240 989067 927484 925080 909988 997705 948045 980596 914840 948292 919109 910112 966308 900727 910960 979205 931854 908333 991209 981839 956166 974613 995203 966990 938910 941216 914672 933355 985615 982778 957381 988742 953389 991108 947903 912472 918163 969906 964111 929962 935305 902548 945714 974104 928615 924305 905236 998020 905486 930862 901235 989849 940154 937148 980056 959679 947583 998613 977242 916833 969282 949170 900368 961342 939999 929601 944787 995662 974028 958310 921584 958575 933567 906280 988512 959178 904590 988676 957739 907989 900661 965456 944873 994471 994972 955959 948627 903685 901345 954396 903299 930719 996009 993267 950060 960314 915834 999382 935434 954938 944206 979618 917776 951187 981313 958163 940639 922630 992305 992097 972307 932736 925670 953918 986614 974594 934660 942494 917284 979430 984737 917590 947140 997383 920975 984872 942670 985233 930377 953830 985906 986053 966726 963423 950922 912475 945103 976719 985123 924887 910775 925249 999588 949391 951667 989175 920498 912843 942159 948885 920259 957780 973376 993195 991822 944663 908246 968531 982387 933242 953096 945323 930179 963710 929505 970332 916922 973184 984611 920329 981018 925582 946488 924923 962373 955257 998001 934710 911726 992542 929291 998178 928355 934427 952582 931030 982605 923690 999752 906602 964817 936319 987505 952081 992811 958425 982864 938916 990323 958446 983302 983941 946765 973179 901160 931711 919599 946767 960459 930851 933541 971103 933435 993107 942027 909167 942562 975877 921428 958979 979338 935557 965507 906258 977238 980549 902691 980024 945202 990599 931617 961529 906478 948920 982345 917331 916748 941358 902246 996764 915497 929991 903399 920238 967440 965281 901481 944954 955117 929995 970612 964013 939907 936178 979668 966600 958241 936029 916484 951743 966042 967614 998074 925877 993390 960251 967925 941807 966017 962232 939261 976755 932140 932715 977158 912581 948651 976450 928286 979040 905094 938335 917477 964001 938970 914038 948331 907626 935049 927409 904651 997034 970275 954674 916073 940371 989140 988121 980567 952227 902612 958615 977667 945311 943859 950217 989587 907699 958898 916108 997831 920592 985994 978237 978276 906123 932824 942960 941819 926844 908357 983601 909853 994213 951052 908867 996782 983303 969154 966484 980194 993199 932232 986659 926894 913074 976814 994008 971119 971678 912191 942954 962042 931554 995207 944311 940910 982507 932929 996526 977343 987425 900264 942522 985107 928795 921491 943306 957562 985766 956808 990110 904290 908820 978493 964362 922627 915361 916232 903914 901593 971136 911829 998776 981211 943156 977154 981499 963855 942809 991125 934839 907915 965588 930265 989006 959590 931308 973149 922061 973052 920895 975676 977277 956149 975657 927962 929591 900437 939042 974841 915788 926732 935559 933884 974538 903399 907525 915684 932315 972467 957880 947938 976670 931951 971514 986902 975633 950223 981755 951300 903062 974021 911159 945098 980546 994678 926166 914377 996272 970796 989754 981804 932192 969130 975940 928097 921430 999114 940843 912725 931833 901264 932569 925315 940405 914529 946256 935615 912770 927191 992686 908408 976341 910220 984645 959613 960621 935876 951380 947886 915297 938511 943397 982458 942738 925282 917813 946688 968483 984004 980446 934707 944365 960940 985234 905824 967174 935060 976099 920189 973014 914809 946068 995858 974572 951471 996020 960505 934099 994171 938061 933736 925278 990976 904749 954985 950486 920425 989603 908267 938566 937248 914198 928911 979906 982915 947309 900157 959518 946194 973678 949582 983617 986995 941943 908807 900644 904703 985761 947036 920704 951537 949872 969660 969597 901814 951493 991249 957259 910407 988015 909753 904080 934773 989681 956657 970951 929664 953437 907490 909409 918538 952261 963418 906806 932506 907506 962079 956611 996095 906014 951381 964345 912016 909895 961672 969823 981552 949037 947934 936448 942032 988444 971337 914069 903196 937667 901037 962803 965952 965666 972913 905064 916455 903424 997797 926765 988549 962164 978372 931115 920517 931849 986889 958289 944152 993411 902556 944998 936960 970424 922927 969366 974320 956226 919204 930514 993743 943129 911282 992730 902549 990251 954642 920887 936843 995958 945341 938007 973661 967740 998504 993381 996494 943372 948187 923031 975694 926261 982275 914350 946857 940991 919194 983083 970714 945823 943680 971737 986666 952980 959454 963381 928908 974626 917629 993943 930732 937750 906784 913958 961951 919724 957003 920773 978064 953319 975234 929073 987565 963906 952667 906512 902547 984916 981025 986499 923227 923296 979063 935513 979382 974261 921899 995865 916183 908966 992655 901765 937172 946964 960068 921462 923492 922101 919035 906519 961269 975397 980054 990583 941470 927804 946415 953620 999409 900216 917162 984067 982722 930532 978165 912684 942539 946363 933553 932350 980202 927555 931310 956686 942409 978957 979023 927495 953256 989708 939599 906165 990080 917450 979539 964153 964811 936101 936299 992582 996789 972394 915597 909076 944928 923442 989752 963858 988144 996029 918351 927280 989436 905400 960255 999432 992258 932965 983571 998374 957359 963260 962979 909835 949698 976493 964330 958752 908579 928846 986074 951065 909313 981787 934341 940210 999678 944255 911075 968537 969866 901456 935043 906720 947413 940695 944148 962643 975348 924837 920829 934665 908630 901033 932921 929877 947808 966290 937062 950221 967791 914966 904766 943371 917815 985412 942847 999814 973679 968274 943857 942507 913259 998878 983473 929176 970306 907645 969748 960595 973628 982604 968954 990399 970991 939806 903258 924149 935669 921494 949446 978093 970688 992091 986889 961869 976982 996534 926516 940155 999142 973328 933035 951454 915132 981693 942383 970171 988341 958928 971273 917194 967072 943825 950597 991636 930944 978479 985608 986003 967297 924953 972867 962931 978757 909807 987802 905219 959685 934645 977139 977216 982691 989371 949931 958141 980257 917605 977989 901958 981131 928992 964980 965305 919145 907866 915785 987535 913345 911782 938533 970900 963310 998117 907679 940671 921240 921898 958974 977874 974256 985645 929205 963718 969144 913238 993524 979820 912293 967208 946806 931146 915572 941923 961891 989492 974059 935638 968614 937065 969648 927678 912413 988792 900456 944687 919444 952773 930121 911522 920462 953756 989804 900785 991033 965664 970236 969415 997635 910298 928388 959936 964149 990118 938473 961200 969426 901822 938846 918651 987645 965520 915735 981758 941882 927841 994306 946420 969683 983957 967127 935884 926018 969822 908786 946179 989688 954262 959706 903828 918478 901912 953475 972461 900906 928807 906941 981216 915291 954763 910361 913775 944599 931781 952521 939896 924720 979711 980819 966138 973628 997658 963662 983327 916613 965668 934534 920146 979993 996749 979044 924621 955622 923111 910306 904970 960642 956599 945005 957540 997531 919317 903764 933208 971110 909454 927847 996497 939413 903474 976981 943806 925530 965755 983273 993929 923122 988202 960045 922226 930193 914859 947770 910216 923147 958835 946112 958739 927413 994512 963325 939667 940825 979576 976687 953151 987533 919161 978605 919989 962065 999264 946033 947915 964414 953002 906892 935450 909231 906636 963354 948413 919796 948960 945162 987673 995284 925348 930632 999675 902123 919383 913531 922853 962879 926590 919640 981763 910889 971611 949929 932444 929375 908310 934580 990562 959300 992553 936594 985844 942267 965626 973714 938703 931179 968661 977782 943033 939439 997301 985666 943500 929431 921010 953795 945293 965635 901942 958614 916157 989120 924453 990912 971959 944879 996602 993971 927742 917606 975349 961836 953058 916157 987034 908199 937365 971606 995153 962938 922290 942991 925395 983376 954716 976858 941928 978543 967130 905098 972172 924030 905562 999921 905953 965744 961605 998674 930352 974213 922363 927728 931641 948978 932924 906408 979820 935002 935042 980520 942163 932332 943471 927307 960618 994190 999426 967083 995924 922106 959693 986899 916087 996636 988865 986376 931713 965757 957586 938698 968717 972539 947916 991507 925768 908306 933752 979901 901115 966240 977442 990243 909056 996021 944013 909558 966457 952701 967867 951612 994641 912504 933415 915150 926483 976131 939317 973593 933734 988221 915848 911918 911530 957058 905620 979935 953470 910289 905628 963090 979850 938691 934671 932985 915235 994529 920083 920175 967750 946855 955477 992093 958034 924116 901950 941657 999183 908163 904470 937233 944786 907693 953969 930206 927036 966404 984370 925545 972235 906685 912879 976369 931461 913504 931438 911621 969722 963586 919203 925930 954852 934281 995540 917633 926695 958097 991695 991201 978551 999366 918476 907027 983675 927589 904036 965676 970828 944559 980593 945762 976500 906110 957870 931572 987466 954660 930062 929309 965424 928989 950390 915092 988733 961824 904731 978348 976293 954911 970929 969504 930805 981926 912226 934428 918649 981869 955637 905938 969287 975416 947741 972810 950057 987196 967801 966611 961207 934615 922657 970828 974299 933264 933473 937348 916260 919451 913988 955348 953367 985418 923886 936996 953407 952642 934209 921294 928047 905007 916743 921652 987487 903870 905426 988183 976276 901284 974159 921407 901541 941512 995183 939711 969898 914685 917240 958616 945513 905593 940183 913461 943967 973922 969221 925880 938353 970454 927748 975881 971705 928193 932631 987418 956434 998168 925938 984467 958045 909709 970764 910164 902339 950413 906454 951248 990129 939437 907301 936475 988286 935792 979961 910956 974479 989586 969032 953482 970402 997831 997716 915385 911466 982244 933037 919481 907516 902798 971218 994856 955232 970155 987279 986685 953333 905829 989719 996358 982878 982916 973062 901672 954254 998882 910716 990930 906888 917945 968912 939599 987479 995419 966755 937678 979268 933998 948189 947661 973608 949320 982100 906552 979511 931234 974439 997683 992361 982391 964067 963776 943127 997408 916272 995971 984978 917772 921958 968922 932216 901249 905122 922678 944538 910333 949562 912785 938772 983683 959399 918136 996338 928647 958093 989463 951759 975001 934313 958721 997412 934968 952352 985584 901285 920079 986561 902065 929717 991898 979251 908016 984015 971894 992013 931361 959636 977818 936283 966233 901924 961216 909045 959820 942931 909983 995043 966901 950484 914414 945670 967388 903401 961684 957587 940490 911383 927417 995293 984390 930358 901250 943914 976709 961707 922551 986505 916400 996693 939334 990870 985769 914715 999005 988798 911880 922038 910173 900974 928491 931197 936129 925079 984186 948301 916454 940224 996475 945345 994806 942270 905736 989274 905527 902505 978599 957815 901539 916688 998131 977206 934589 983997 983849 949558 974003 974772 968049 965163 980283 981623 921781 932028 995071 988813 921580 933219 966550 953865 984333 926500 952752 926703 994196 943636 918419 935033 966511 900141 918458 901967 903887 992202 929256 906657 950118 915225 991733 933995 914703 942660 974728 945662 932295 957650 978734 919887 903375 941343 975401 903587 918749 929846 993365 992864 911969 935557 982695 946674 993698 965323 942344 985090 904803 970699 934182 905966 929680 979524 906278 976009 989515 923704 946355 944423 973858 930745 963688 925067 975277 982184 998181 933293 947275 961422 964168 936162 905856 948878 924980 995853 923015 959087 941829 943676 985031 934962 992971 975184 919432 924536 918033 969957 932701 902907 901845 998587 981328 911391 978089 987631 985781 939811 940037 911213 912584 987816 912097 928742 948939 917385 931667 985318 942658 907745 924472 928260 919831 939121 976564 947783 910820 936942 972237 942484 996710 976080 924030 924668 927604 932247 946462 940714 988731 909033 932729 927269 908965 924093 934067 918658 977891 998629 924661 963544 926868 982301 938802 924842 900998 992619 930640 969615 989007 963483 962116 981155 962506 978511 916124 907129 933199 977242 976890 977328 921812 925294 944436 957725 925671 917443 964767 976429 952963 936824 977969 972009 973181 911394 913419 965691 939517 987109 961747 957471 981070 974103 956718 905116 941194 992585 905140 917295 988693 957456 965054 901688 967928 948767 904864 970502 971506 988756 951352 977872 998175 963856 947594 903291 973746 966792 964981 924860 977809 990664 914375 903097 901033 939477 993435 925271 979741 994124 970334 987752 936572 905545 954794 985043 969264 983578 958253 905621 998281 956952 999948 978446 937779 945344 904464 973524 931622 958533 912741 976858 923697 955774 973447 913957 933035 901601 965640 945280 916185 998577 932836 918141 953549 976846 950209 990371 964479 959622 913156 900016 937559 980779 967781 954791 989498 906265 996674 971440 906921 930979 964414 957743 977479 935012 982123 994638 920147 929819 966303 986171 909989 976516 916559 911430 922276 937801 948374 930545 937248 943225 954761 918963 961146 964250 911476 984579 901257 950335 949515 963670 914749 916779 907246 937525 932936 978466 954853 993548 955058 975667 945830 948765 964838 981483 907978 904323 945028 953286 947151 977190 919032 900558 953429 997226 917057 903383 933322 947903 979943 977076 939384 917162 938950 903620 955756 915185 952880 976034 965400 905896 971724 920289 909767 907532 920911 936185 952008 984087 918249 970194 958458 980764 921279 942771 972670 913736 997003 994423 919502 942049 959987 971405 906545 919720 986244 928937 961152 946254 951256 966032 997954 906665 955503 943020 982142 949861 996644 916539 912603 946191 949617 907589 905404 976589 934811 989624 915260 978599 904262 984017 912442 950038 997656 958771 921476 930151 916867 984186 938163 904933 928506 985542 991967 964643 985288 931322 959860 962443 940804 986352 905668 925362 994518 963822 991817 929554 935893 971926 925173 972858 969041 963212 906392 961881 958046 995028 943491 904306 938795 936731 920264 903671 942555 962628 982049 991244 987491 988900 954594 946428 949632 900749 916358 933280 953660 901776 978412 988560 936596 927024 988430 932562 954965 949880 930349 901343 968675 926514 992034 960911 903460 906943 915802 917702 952174 901704 994381 987498 934875 970759 903331 976130 989123 935468 919357 984729 991148 918465 906411 964125 928496 974007 926478 985696 949817 937509 938185 953194 936829 939187 978565 945787 948406 971187 920475 952491 920931 968240 913840 946408 959797 929504 921977 974540 966358 984291 908807 916344 953602 935179 935144 973783 941596 961271 981206 944855 904443 977356 994457 901134 951570 957327 964734 992241 915008 912228 903246 909562 914140 914262 944854 957246 958596 941292 907345 960313 916775 964756 994516 996472 958497 939837 906893 931352 914598 957567 962503 970810 902521 932075 994279 951667 976455 964995 993932 912404 966558 984097 969308 910647 1
