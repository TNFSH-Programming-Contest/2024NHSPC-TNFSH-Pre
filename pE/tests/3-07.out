2228925121614472